Sources/MongoSwift/Operations/FindOperation.swift

```diff
@@ -182,6 +182,23 @@ public struct FindOptions: Codable {
         self.sort = sort
     }
 
+    internal init(findOneOptions: FindOneOptions) {
```

Comment:
I think this initializer also needs to copy over the read preference. I think we should probably also test using a read preference with this method doesn't error in `ReadPreferenceTests.swift`. 

this makes me realize we don't really have a good way right now of ensuring specified read preferences are actually sent to / respected by libmongoc. one way we might be able to test that is by using an unsatisfiable read preference (e.g. containing tags that match no members in the replica set) and ensuring commands fail.... what do you think @patrickfreed ?

I think that's out of scope for this ticket but something we should think about. I've opened [SWIFT-685](https://jira.mongodb.org/browse/SWIFT-685) for discussion.

Reply:
That sounds like a reasonable approach 👍 

With regards to the naming of this, our conversion initializerss generally follow the form of `init(from:)`, so this could be updated to match that (e.g. see [`UUID.init(from)`](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/BSON/BSONValue.swift#L219))